Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/ScanAPIModelWizardPage.xaml.cs

```diff
@@ -284,7 +357,102 @@ public class SoapData
+            if (deltaAPI != null)
+            {
+                bool showMergerWindow = false;
+                if (deltaAPI.DefaultOperationEnum == DeltaAPIModel.eHandlingOperations.MergeChanges)
+                    showMergerWindow = true;
+
+                if (mergerWindow == null)
+                    mergerWindow = new MergerWindow(deltaAPI);
+
+                mergerWindow.ownerWindow = (Window)AddAPIModelWizard.mWizardWindow;
+                mergerWindow.ShowAsWindow(showMergerWindow);
+            }
+        }
+
+        private void XHandlingOperation_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            ComboBox handlingOpCB = sender as ComboBox;
+            string mergerDescription = Convert.ToString(DeltaAPIModel.GetEnumDescription(DeltaAPIModel.eHandlingOperations.MergeChanges));
+
+            DeltaAPIModel.eHandlingOperations selectedOperation = DeltaAPIModel.GetValueFromDescription<DeltaAPIModel.eHandlingOperations>(handlingOpCB.SelectedItem.ToString());
+
+            DeltaAPIModel deltaAPI = null;
+            var fEl = sender as FrameworkElement;
+            if (fEl != null)
+                deltaAPI = fEl.DataContext as DeltaAPIModel;
+            if (deltaAPI != null)
+            {
+                deltaAPI.defaultOperation = Convert.ToString(handlingOpCB.SelectedItem);
+
+                // Update default Operation Enum field
+                deltaAPI.DefaultOperationEnum = selectedOperation;
+                if (selectedOperation == DeltaAPIModel.eHandlingOperations.DoNotAdd)
+                    deltaAPI.IsSelected = false;
+                else
+                    deltaAPI.IsSelected = true;
+            }
+
+            // Launch the Merger Window as Merge Changes is selected.
+            if (handlingOpCB.SelectedItem.Equals(mergerDescription))
+            {
+                ShowMergerWindow(sender);
+            }
+        }
+
+        private void XManualMatchBtn_Click(object sender, RoutedEventArgs e)
+        {
+            DeltaAPIModel deltaAPI = null;
+            var fEl = sender as FrameworkElement;
+            if (fEl != null)
+                deltaAPI = fEl.DataContext as DeltaAPIModel;
+
+            if (deltaAPI != null)
+            {
+                SingleItemTreeViewSelectionPage apiModelTreeSelectionPage = null;
+                AppApiModelsFolderTreeItem apiRoot = new AppApiModelsFolderTreeItem(WorkSpace.Instance.SolutionRepository.GetRepositoryItemRootFolder<ApplicationAPIModel>());
+
+                apiModelTreeSelectionPage = new SingleItemTreeViewSelectionPage("API Models", eImageType.APIModel, apiRoot, SingleItemTreeViewSelectionPage.eItemSelectionType.Single, true,
+                                                                                    new System.Tuple<string, string>(nameof(ApplicationAPIModel.EndpointURL), deltaAPI.learnedAPI.EndpointURL));
+
+                if (deltaAPI.learnedAPI.APIType == ApplicationAPIUtils.eWebApiType.SOAP)
```

Comment:
when user is doing manual match- filter the API models he see only by type Soap/Rest- let him do the rest, because if not then we might filter API he want to select